### PR TITLE
Code quality & no matrix_id/_index if no matrix

### PR DESCRIPTION
**Pull request by senhalil**

### CODE REVIEW

**Review by braktar — CHANGES_REQUESTED** [review]
test?

**Comment by senhalil** [comment]
@braktar  you can review again, the code quality changes are included as well.

**Review by braktar — APPROVED** [review]

File: models/vrp.rb [hunk 1]
```diff
@@ -135,17 +135,37 @@ def self.create(hash, delete = true)
     end
 
     def self.check_consistency(hash)
-      periodic = hash[:configuration] && hash[:configuration][:preprocessing] && hash[:configuration][:preprocessing][:first_solution_strategy].to_a.include?('periodic')
+      # matrix_id consistency
+      hash[:vehicles]&.each{ |v|
+        if v[:matrix_id] && (hash[:matrices].nil? || hash[:matrices].none?{ |m| m[:id] == v[:matrix_id] })
+          raise OptimizerWrapper::DiscordantProblemError, 'There is no matrix with id vehicle[:matrix_id]'
```
Comment (fab-girard) [inline models/vrp.rb:141]: rescue this error for a 400 status code

Comment (senhalil) [reply models/vrp.rb:141]: Error codes are fixed.

**Review by braktar — APPROVED** [review]

**Pull request merged**